presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -254,9 +281,14 @@ public Page getNextPage()
         Block[] blocks = new Block[outputColumns.size()];
         for (int i = 0; i < outputColumns.size(); i++) {
             int columnIndex = outputColumns.get(i);
-            Block block = getStreamReaders()[columnIndex].getBlock(positionsToRead, positionCount);
-            updateMaxCombinedBytesPerRow(columnIndex, block);
-            blocks[i] = block;
+            if (constantValues[columnIndex] != null) {
+                blocks[i] = RunLengthEncodedBlock.create(columnTypes.get(columnIndex), constantValues[columnIndex], batchSize);
+            }
+            else {
+                Block block = getStreamReader(columnIndex).getBlock(positionsToRead, positionCount);
+                updateMaxCombinedBytesPerRow(columnIndex, block);
```

Comment:
Sorry for commenting on a very old PR.  should the columnIndex here be hiveColumnIndices[columnIndex]. I got a crash with the following call stack.

```
com.facebook.presto.spi.PrestoException: Failed to read ORC file: ws://wsXXXXXXXX
	at com.facebook.presto.hive.orc.OrcSelectivePageSource.getNextPage(OrcSelectivePageSource.java:106)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.processPageSource(ScanFilterAndProjectOperator.java:277)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:242)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:418)
	at com.facebook.presto.operator.Driver.lambda$processFor$9(Driver.java:301)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:722)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:294)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:599)
	at com.facebook.presto.$gen.Presto_0_257_df401fd__0_257____20210728_224554_1.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 5 out of bounds for length 5
	at com.facebook.presto.orc.AbstractOrcRecordReader.updateMaxCombinedBytesPerRow(AbstractOrcRecordReader.java:397)
	at com.facebook.presto.orc.OrcSelectiveRecordReader.getNextPage(OrcSelectiveRecordReader.java:725)
	at com.facebook.presto.hive.orc.OrcSelectivePageSource.getNextPage(OrcSelectivePageSource.java:86)
	... 13 more
```

Reply:
@arunthirupathi Indeed. Looks like we should use hiveColumnIndices[columnIndex] here (just like we do in getStreamReader:

```
    private SelectiveStreamReader getStreamReader(int columnIndex)
    {
        return getStreamReaders()[hiveColumnIndices[columnIndex]];
    }
```

Do you think you can create a test case to reproduce this issue?

Reply:
I will create a PR with the test to repro this.